### DISCUSSION TITLE
Need help typing `getElements` return value in typescript

### DISCUSSION THREAD

**Discussion by sarahandromeda** [webdriverio/webdriverio#11989] [Q&A] [answered]
Hello all!

I have a really basic question but I just can not figure it out on my own. I am using `Webdriverio` via Typescript in a project. I am using the shorthand of  `getElements` to retrieve a list of elements and then pull the last element from the list and get it's text value.

For example (i know 0 doesn't pull the last item, this is just an example):
```
let foundElems = await $$(someSelector);
// getText method doesn't exist unless I get the correct type
foundElems[0].getText();
```

However, because I am using typescript, I need a type for `foundElems`. I wasn't able to find a type in the docs other than `ChainablePromiseArray` which isn't a viable type once all the promises resolve (i think, either way it does not work as a type in this case). I found online that the return value should be a `WebElement` but I am unable to import that class/type. I could not find that type/class anywhere in my installed webdriver modules or types.

I did make sure to follow along with the [Typescript Setup in the Docs](https://v5.webdriver.io/docs/typescript.html) to set up my install correctly but I still can not find the correct type. I am not sure if the information I found on `WebElement` is outdated, but looking through the types provided in `@wdio/types` I wasn't able to find anything that looked like the correct type.

Does anyone know the return value for `getElements` (`$$()`) and where I can import it from? And if it is meant to be `WebElement` does anyone know where I can find it to import?

Any help is greatly appreciated!! Thank you!

**Comment by christian-bromann** [comment, marked as answer]
@sarahandromeda the return value for `await $$(someSelector)` should be `WebdriverIO.ElementArray` and `foundElems[0]` should resolve to `WebdriverIO.Element`. If you don't have the `WebdriverIO` namespace available, I guess the type setup is not correct. It should normally pick it up automatically. Try one of the following:

- if you rely on globals, make sure to set `@wdio/globals/types` to the `tsconfig.json` as you can see in the docs
- import globals directly via `import { $$ } from '@wdio/globals'`
- Try to add `/// <reference types="webdriverio">` to the top of the file.

Let me know if these work.

**Reply by sarahandromeda** [reply]
Thank you for the quick reply! I'm sorry I am  just getting to this now (this was an issue I had at work and it's been a busy day). Let me try that and get back to you momentarily.

**Reply by sarahandromeda** [reply]
Aha!! Importing the globals directly did the trick(second bullet point). I had bullet point 1 done but when running `getText` on the element, it said the function did not exist which is why I needed to define a type. 
However, after adding the `import { $$ } from '@wdio/globals'` line, I was able to develop the code without worrying about the typing.
```
import { $$ } from '@wdio/globals'
// No type needed here, inferred from $$ import from global
let foundElems = await $$(someSelector);
let someElem = foundElems[0];
// Now `getText` exists as expected
console.log(someElem.getText());
```
Thank you so much, I truly appreciate the time and the help!!
